libsmack/common.c

```diff
@@ -204,3 +205,64 @@ int apply_cipso(const char *path)
 
 	return 0;
 }
+
+int apply_onlycap(const char *path)
+{
+	FILE *file = fopen(path, "r");
+	int ret = 0;
+	if (file == NULL) {
+		if (errno == ENOENT) /* there is no configuration file,
+				        we should do nothing */
+			return 0;
+		fprintf(stderr, "open() failed for '%s' : %s\n", path,
+			strerror(errno));
+		return -1;
+	}
+
+	char buf[SMACK_LABEL_LEN + 2];
+	int cnt = 0;
+	int size = 10;
+	char **labels = malloc(sizeof(char *) * size);
+	if (labels == NULL) {
+		fputs("Out of memory.\n", stderr);
+		ret = -1;
+		goto out;
+	}
+
+	while (fgets(buf, SMACK_LABEL_LEN + 2, file) != NULL) {
+		if (strcmp(buf, "\n") == 0)
+			continue;
+		if (cnt == size) {
+			size = size * 2;
+			char **new_labels = realloc(labels, sizeof(char *) * size);
+			if (new_labels == NULL) {
+				fputs("Out of memory.\n", stderr);
+				ret = -1;
+				goto out;
+			}
+			labels = new_labels;
+		}
+		int label_len = strlen(buf);
+		char* label = malloc(label_len + 1);
+		if (label == NULL) {
+			fputs("Out of memory.\n", stderr);
+			ret = -1;
+			goto out;
+		}
+		memcpy(label, buf, label_len + 1);
+		if (get_label(label, buf, NULL) <= 0) {
```

Comment:
This does not compile. The `get_label` function is declared as static inline in libsmack.c and is not available in common.c.